test/helpers.bash

```diff
@@ -0,0 +1,236 @@
+#  run_in_host_netns  #  Run args in host netns
+################
+#
+function run_in_host_netns() {
+    run_helper nsenter -n -t $HOST_NS_PID "$@"
+}
+
+#### Functions below are taken from podman and adapted to netavark.
+
+################
+#  run_helper  #  Invoke args, with timeout, using BATS 'run'
+################
+#
+# This is the preferred mechanism for invoking podman: first, it
+# invokes $NETAVARK, which may be 'netavark' or '/some/path/netavark'.
+#
+# Second, we use 'timeout' to abort (with a diagnostic) if something
+# takes too long; this is preferable to a CI hang.
+#
+# Third, we log the command run and its output. This doesn't normally
+# appear in BATS output, but it will if there's an error.
+#
+# Next, we check exit status. Since the normal desired code is 0,
+# that's the default; but the first argument can override:
+#
+#     run_netavark 125  nonexistent-subcommand
+#     run_netavark '?'  some-other-command       # let our caller check status
+#
+# Since we use the BATS 'run' mechanism, $output and $status will be
+# defined for our caller.
+#
+function run_helper() {
+    # expected_rc if unset set default to 0
+    expected_rc="${expected_rc-0}"
+    if [ "$expected_rc" == "?" ]; then
+        expected_rc=
+    fi
+    # Remember command args, for possible use in later diagnostic messages
+    MOST_RECENT_COMMAND="$*"
+
+    # stdout is only emitted upon error; this echo is to help a debugger
+    echo "$_LOG_PROMPT $*"
+
+    # BATS hangs if a subprocess remains and keeps FD 3 open; this happens
+    # if a process crashes unexpectedly without cleaning up subprocesses.
+    run timeout --foreground -v --kill=10 10 "$@" 3>/dev/null
+    # without "quotes", multiple lines are glommed together into one
+    if [ -n "$output" ]; then
+        echo "$output"
+    fi
+    if [ "$status" -ne 0 ]; then
+        echo -n "[ rc=$status "
+        if [ -n "$expected_rc" ]; then
+            if [ "$status" -eq "$expected_rc" ]; then
+                echo -n "(expected) "
+            else
+                echo -n "(** EXPECTED $expected_rc **) "
+            fi
+        fi
+        echo "]"
+    fi
+
+    if [ "$status" -eq 124 ]; then
+        if expr "$output" : ".*timeout: sending" >/dev/null; then
+            # It's possible for a subtest to _want_ a timeout
+            if [[ "$expected_rc" != "124" ]]; then
+                echo "*** TIMED OUT ***"
+                false
+            fi
+        fi
+    fi
+
+    if [ -n "$expected_rc" ]; then
+        if [ "$status" -ne "$expected_rc" ]; then
+            die "exit code is $status; expected $expected_rc"
+        fi
+    fi
+
+    # unset
+    unset expected_rc
+}
+
+#########
+#  die  #  Abort with helpful message
+#########
+function die() {
+    # FIXME: handle multi-line output
+    echo "#/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv" >&2
+    echo "#| FAIL: $*" >&2
+    echo "#\\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^" >&2
+    false
+}
+
+########
```

Comment:
Would you consider using `assert` and `expect_output` from [buildah:helpers.bash](https://github.com/containers/buildah/blob/68cd60292259fa6b9fe09b9eee06d391eaf8228a/tests/helpers.bash#L252) instead of from podman? Podman tests came first; when I worked on Buildah I had learned a lot from my mistakes. Those are better. (And yes, I always meant to backport, but no, it's probably too late to do so).

Reply:
Is there any point in using `expect_output`, it looks like assert is enough?

Reply:
It's a minor shortcut for what is by far the most common expression.
```bash
    assert "$output" = "foo bar" "test failed"
    expect_output "foo bar" "test failed"
```
There's just a tiny bit less chance of screwing it up using the second form; and fewer quotation marks makes it slightly more readable. But it's personal preference, really, and I can see a benefit to using only `assert`.